### ISSUE TITLE
When RHS object name is already a name in the table

### ISSUE THREAD

**Issue by KyleHaynes** [Rdatatable/data.table#5609]
Raised this with @MichaelChirico on Twitter, and he said to post.

When the RHS object name is already another column in the `data.table` being evaluated, an error is thrown.

@MichaelChirico pointed to:
https://t.co/DgVJEuTImL
With comment:

> What's happening is we're intercepting %in% and trying to turn it into a join subset. There's some logic around ensuring the LHS of %in% is a symbol, but not the RHS. We end up trying to evaluate 'x$x' _within_ the table, i.e. 'x-the-column'$x, not 'x-the-table'$x.

Repex:

```R
require(data.table)
x = data.table(x = 1:3, z = 1:3)
y = data.table(x = 1:3)
z = data.table(x = 1:3, z = 1:3)


## Code Fail ----
y[x %in% x$x]
# Error in x$x : $ operator is invalid for atomic vectors
y[x %in% x[["x"]]]
# Error in x[["x"]] : subscript out of bounds

y[x %in% z$x]
#    x
# 1: 1
# 2: 2
# 3: 3

y[x %in% y[["x"]]]
#    x
# 1: 1
# 2: 2
# 3: 3
```

**Comment by tdhock** [comment]
it seems to me that this behavior is normal and should not be changed
(although it may be an expected for some users and so maybe we can add Some logic that looks for an object with that name in the environment, and if there is one than a helpful an error message that says to use a join instead)

**Comment by KharuRai** [comment]
Hello @tdhock I would like to work on this issue.

**Comment by Mukulyadav2004** [comment]
Hi @tdhock 
I believe the fix is changing `RHS = eval(stub[[3L]], x, enclos)` to `RHS = eval(stub[[3L]], enclos)`, which forces the RHS to evaluate in the parent frame. 
Please let me know if I'm right, If so, may I proceed with a PR.

**Comment by tdhock** [comment]
no, the current behavior is correct, but we could improve the error message.

**Comment by tdhock** [comment]
on second thought, it is normal to have the same error message as base R, I don't think we need to improve the error message, closing.
